eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Epoch.hs

```diff
@@ -136,6 +149,37 @@ instance
   type Event (ConwayEPOCH era) = ConwayEpochEvent era
   transitionRules = [epochTransition]
 
+returnProposalDeposits' ::
+  ConwayGovernance era ->
+  StrictSeq (GovernanceActionId (EraCrypto era), GovernanceActionState era) ->
+  UMap (EraCrypto era) ->
+  UMap (EraCrypto era)
+returnProposalDeposits' ConwayGovernance {..} gaids m =
+  RewDepUView m ∪+ foldl' addRew mempty gaids
```

Comment:
`unionRewAgg`/`∪+` has an invariant:
> We presume that the domain of the `Map` on the right, is a subset of the domain of the `UView` on the left.

Which means if the reward account is removed from UMap before the epoch boundary or not even present to begin with in the UMap, then the deposit will be dropped on the floor.

I might be wrong of course, but I think it is a valid concern.

Reply:
good catch, we have a similar operator in the spec which acts as `unionWith`, so that's why I made that mistake here